### PR TITLE
Pluralizes forwardTo field of TCPRoute

### PR DESCRIPTION
**Pull request by danehans**
Similar to https://github.com/kubernetes-sigs/service-apis/pull/57, this PR pluralizes the `ForwardTo` field of the `TCPRouteAction` API type so traffic splitting can be supported by the `tcproutes` resource.

/assign @robscott @bowei 
/cc @hbagdi @JeremyOT @jpeach @Miciah @howardjohn

### CODE REVIEW

File: apis/v1alpha1/forwarding_target_types.go [hunk 1]
```diff
@@ -59,7 +59,8 @@ type ForwardToTarget struct {
 	// If only one targetRef is specified, 100% of the traffic is forwarded to the
 	// targetRef. If unspecified, weight defaults to 1.
 	//
-	// Support: Core
+	// Support: core (httproute)
+	// Support: extended (tcproute)
```
Comment (jpeach) [inline apis/v1alpha1/forwarding_target_types.go:63]: Nit, probably should capitalize "Core" and "Extended"

File: apis/v1alpha1/tcproute_types.go [hunk 2]
```diff
@@ -56,15 +56,18 @@ type TCPRouteRule struct {
...
 	// Omitting or specifying the empty string for both the resource and
 	// group indicates that the resource is "services".  If the referent
 	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the TCPRoute will be true.
-	ForwardTo *ForwardToTarget `json:"forwardTo" protobuf:"bytes,1,opt,name=forwardTo"`
+	// that includes the HTTPRoute will be true.
+	//
+	// Support: core
+	//
+	ForwardTo []ForwardToTarget `json:"forwardTo" protobuf:"bytes,1,rep,name=forwardTo"`
```
Comment (jpeach) [inline apis/v1alpha1/tcproute_types.go:70]: suggest (here and in HTTPRouteAction)
```
// +kubebuilder:validation:MinItems=1
```

**Comment by robscott** [comment]
Thanks!

/lgtm

**Pull request merged**